### PR TITLE
Improved the Create Assignment page

### LINKED ISSUE #555: Unequal width and length of Input Fields
**To Reproduce**
Can be found on http://localhost:8080/groups/1/assignments/new

**Screenshots**
Issue

Also, unequal width of input fields for "Name", "Grading Scale" and "Deadline"

### LINKED ISSUE #652: Improve Create Assignment page
**Is your feature request related to a problem? Please describe.**
The Create Assignment page is not user friendly and can be improved.

**Describe the solution you'd like**
* First of all we can change the way we set the deadline, it can either be a calendar to set the date or a clock to set the duration.
* We can remove the "help" link and write the info there.
* Also align the **Cannot be changed once set** warning with the appropriate field. It isn't clear if it is for the name or grade.
* Favicon vanishes on reloading the page.

**Are you working on this? (Yes/No)**
Nope. For GCI 2019

### PR DESCRIPTION
**Pull request by MrBartusek**
Fixes #555
Fixes #652
### Screenshots of the changes (If any) -

Before my changes:
image

After: (code is cleaner too)
image

This is a [GCI Task](https://codein.withgoogle.com/dashboard/task-instances/5182641653940224/)

### CODE REVIEW

**Comment by MrBartusek** [comment]
hound errors about ES6 and codeclimate errors about functions with `_` were discussed on [Slack](https://circuitverse-team.slack.com/archives/CQFAC5L7P/p1577803381000700)

File: public/js/restrictElements.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        const category = Object.entries(elementHierarchy)[i];
+        const html = _generateRow(category[0], category[1], restrictionMap);
+        $('.restricted-elements-list').append(html);
+    }
+}
+
+function loadRestrictions(restrictions) {
+    _handleMainCheckbox();
+
+    const elementHierarchy = metadata.elementHierarchy;
```
Comment (JoshHeng) [inline public/js/restrictElements.js]: Use `const { elementHierarchy } = metadata;` instead

**Comment by MrBartusek** [comment]
@JoshHeng that is Way to Mark functions private cause we don't use modules

**Comment by nitin10s** [comment]
What fonts are you using? What about the UI of rows: Inputs, Outputs, Gates ...
Also, change the name input field to the one in this:https://xd.adobe.com/spec/7453ce8f-6d79-4969-4a1a-d0162c42c53d-0262/screen/8ba805b4-c7b7-4f05-85eb-52c1a9983ecd/Login [For Consistency]
At last, increase the margin between navbar and the new assignment div.

The UI now looks much better and clean. Good work!

File: public/js/restrictElements.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+function _handleMainCheckbox() {
```
Comment (tachyons) [inline public/js/restrictElements.js]: Do not add anymore classess in public/js, add in app/assets/javascript

**Comment by tachyons** [comment]
Remove undercore from method name prefix since we aren't following that convention

**Pull request merged**